src/Tablebot/Plugin/Discord.hs

```diff
@@ -94,3 +126,15 @@ getMessageMember m = gMM (messageGuild m) m
     gMM (Just g') m' = do
       a <- lift $ restCall $ R.GetGuildMember g' (userId $ messageAuthor m')
       return $ maybeRight a
+
+-- | @toMention@ converts a user to its corresponding mention
+toMention :: User -> Text
+toMention u = pack $ toMentionStr u
+
+-- | @toMentionStr@ converts a user to its corresponding mention, returning a string to prevent packing and unpacking
+toMentionStr :: User -> String
+toMentionStr u = "<@!" ++ show (userId u) ++ ">"
+
+getMessageLink :: Maybe GuildId -> Maybe ChannelId -> Maybe MessageId -> String
+getMessageLink (Just g) (Just c) (Just m) = "https://discord.com/channels/" ++ show g ++ "/" ++ show c ++ "/" ++ show m
+getMessageLink _ _ _ = ""
```

Comment:
I don't have a solution for this, but I wonder if the `Maybe x -> Maybe y -> Maybe z -> t` patten could be generally avoided? (It's been used fairly frequently.) Part of me thinks that applicatives help - you may get some use out of `liftA3` and friends?